cloud/satellite.go

```diff
@@ -74,21 +95,47 @@ func (c *client) LaunchSatellite(ctx context.Context, name, orgID string, featur
 	return nil
 }
 
+func (c *client) ReserveSatellite(ctx context.Context, name, orgID string, out chan<- string) error {
+	defer close(out)
+	ctxTimeout, cancel := context.WithTimeout(ctx, 5*time.Minute)
+	defer cancel()
+	stream, err := c.pipelines.ReserveSatellite(c.withAuth(ctxTimeout), &pb.ReserveSatelliteRequest{
+		OrgId: orgID,
+		Name:  name,
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed opening satellite reserve stream")
+	}
+	var update *pb.ReserveSatelliteResponse
+	for {
+		update, err = stream.Recv()
+		if err == io.EOF {
+			return nil
```

Comment:
if update was ever set to `SatelliteStatusFailed`, it seems like the reserve call failed, and returning `nil` as the error could be confusing.

Can we do an explicit check for this sending it to the `out` channel, and return an error instead?

Reply:
The failure status was added initially when we started satellites, but it was never used anywhere. I'm torn between marking it as deprecated and leaving it as supported in case there's a use for it eventually (e.g. perhaps we'll implement health checks and then can show this state if that fails)

Nonetheless, I don't think there's any harm in adding that check here as precaution in case the API starts returning that status here one day. I'll add it in.